vkcs/resource_vkcs_images_image.go

```diff
@@ -219,6 +221,62 @@ func resourceImagesImage() *schema.Resource {
 	}
 }
 
+type imageCreateError struct {
+	msg string
+}
+
+type imagePrepareUploadError struct {
+	msg string
+}
+
+type imageUploadError struct {
+	msg string
+}
+
+func (e *imageCreateError) Error() string        { return e.msg }
+func (e *imagePrepareUploadError) Error() string { return e.msg }
+func (e *imageUploadError) Error() string        { return e.msg }
+
+func imageCreateAndUpload(imageClient *gophercloud.ServiceClient, d *schema.ResourceData, createOpts *images.CreateOpts) (string, string, error) {
+	log.Printf("[DEBUG] Create Options: %#v", createOpts)
+	newImg, err := images.Create(imageClient, createOpts).Extract()
+	if err != nil {
+		return newImg.ID, "", &imageCreateError{fmt.Sprintf("Error creating Image: %s", err)}
+	}
+
+	var fileChecksum string
+
+	// variable declaration
+	var imgFilePath string
+	var fileSize int64
+	var imgFile *os.File
+
+	// downloading/getting image file props
+	imgFilePath, err = resourceImagesImageFile(imageClient, d)
+	if err != nil {
+		return newImg.ID, "", &imagePrepareUploadError{fmt.Sprintf("Error opening file for Image: %s", err)}
+	}
+	fileSize, fileChecksum, err = resourceImagesImageFileProps(imgFilePath)
+	if err != nil {
+		return newImg.ID, fileChecksum, &imagePrepareUploadError{fmt.Sprintf("Error getting file props: %s", err)}
+	}
+
+	// upload
+	imgFile, err = os.Open(imgFilePath)
+	if err != nil {
+		return newImg.ID, fileChecksum, &imagePrepareUploadError{fmt.Sprintf("Error opening file %q: %s", imgFilePath, err)}
+	}
+	defer imgFile.Close()
+	log.Printf("[WARN] Uploading image %s (%d bytes). This can be pretty long.", d.Id(), fileSize)
+
+	res := imagedata.Upload(imageClient, newImg.ID, imgFile)
+	if res.Err != nil {
```

Comment:
could we be more selective with the error to skip cases like 4xx, 503 etc

Reply:
Expanded error handling to retry creation only if we got 500 error